ts/Core/Axis/TreeGridAxis.ts

```diff
@@ -479,15 +486,25 @@ namespace TreeGridAxis {
 
                     // Update yData now that we have calculated the y values
                     axis.series.forEach(function (series: Highcharts.Series): void {
-                        var data = (series.options.data || []).map(function (
+                        var axisData = (series.options.data || []).map(function (
                             d: Highcharts.PointOptionsType
                         ): Highcharts.PointOptionsType {
-                            return isObject(d, true) ? merge(d) : d;
+
```

Comment:
Where did returning `merge(d)` go? The original function made sure to return a copy of `d` if `d` was an object...